strax/context.py

```diff
@@ -1014,6 +965,50 @@ def get_iter(self, run_id: str,
             raise ValueError(f"Invalid time range: {time_range}, "
                              "returned no chunks!")
 
+    def _make_progress_bar(self, run_id, targets, progress_bar=True):
+        """
+        Make a progress bar for get_iter
+        :param run_id, targets: run_id and targets
+        :param progress_bar: Bool weither or not to display the progress bar
+        :return: progress bar, t_start (run) and t_end (run)
+        """
+        try:
+            t_start, t_end, = self.estimate_run_start_and_end(run_id, targets)
+        except (AttributeError, KeyError, IndexError):
+            # During testing some thing remain a secret
+            t_start, t_end, = 0, float('inf')
+        if t_end == float('inf'):
+            progress_bar = False
+
+        # Define nice progressbar format:
+        bar_format = ("{desc}: "  # The loading plugin x
+                      "|{bar}| "  # Bar that is being filled
+                      "{percentage:.2f} % "  # Percentage
+                      "[{elapsed}<{remaining}]"  # Time estimate
+                      "{postfix}"  # Extra info
+                      )
+        pbar = tqdm(total=1,
+                    desc=f'Loading {targets}',
+                    bar_format=bar_format,
+                    leave=True,
+                    disable=not progress_bar)
```

Comment:
Nice, is this kexword-option new? It definitely makes things simpler. I may have overlooked it.

Reply:
I'm not sure, but indeed it's what made the biggest difference

Reply:
We by the way can safely agree on saying that tqdm is a horribly confusingly documented package. There are so many incompatible statements on this package that it's sometimes complete voodoo.

Reply:
Yes, voodoo is the correct word I guess :D